spec/ngsild_csource_schema.json

```diff
@@ -0,0 +1,137 @@
+                    "items": {
+                        "$ref": "ngsild_schema.json#/definitions/TimeInterval"
+                    }
+                },
+                "location": {
+                    "$ref": "http://json.schemastore.org/geojson#/definitions/geometry"
+                },
+                "expires": {
+                    "type": "string",
+                    "format": "date-time"
+                },
+                "description": {
+                    "type": "string"
+                }
+            }
+        },
+        "CsourceRegistrationMandatoryParams": {
+            "type": "object",
+            "required": [
+                "endpoint",
+                "type",
+                "information"
+            ],
+            "properties": {
+                "endpoint": {
+                    "type": "string",
+                    "format": "uri"
+                },
+                "type": {
+                    "type": "string",
+                    "enum": [
+                        "CSourceRegistration"
+                    ]
+                },
+                "information": {
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                        "$ref": "ngsild_schema.json#/definitions/RegistrationInfo"
+                    }
+                }
+            }
+        },
+        "CsourceRegistration": {
+            "allOf": [
+                {
+                    "type": "object",
+                    "required": [
+                        "id"
+                    ],
+                    "properties": {
+                        "id": {
+                            "type": "string",
+                            "format": "uri"
+                        }
+                    }
+                },
+                {
+                    "$ref": "#/definitions/CsourceRegistrationParams"
+                },
+                {
+                    "$ref": "#/definitions/CsourceRegistrationMandatoryParams"
+                }
+            ]
+        },
+        "CsourceRegistrationDescription": {
+            "allOf": [
+                {
+                    "type": "object",
+                    "properties": {
+                        "id": {
+                            "type": "string",
+                            "format": "uri"
+                        }
+                    }
+                },
+                {
+                    "$ref": "#/definitions/CsourceRegistrationParams"
+                },
+                {
+                    "$ref": "#/definitions/CsourceRegistrationMandatoryParams"
+                }
+            ]
+        },
+        "CsourceRegistrationFragment": {
+            "allOf": [
+                {
+                    "type": "object",
+                    "required": [
+                        "id"
+                    ],
+                    "properties": {
+                        "id": {
```

Comment:
do we really need to duplicate the definitions here in the fragment?

Reply:
Yes from my point of view, since these definitions are not required/mandatory on fragments. However are required during the registration and hence on the description received during the query for a context source registration.